qiskit/compiler/assemble.py

```diff
@@ -16,128 +16,136 @@
 from qiskit.assembler import assemble_circuits, assemble_schedules
-from qiskit.qobj import QobjHeader
+from qiskit.qobj import QobjHeader, Qobj
 from qiskit.qobj.utils import MeasLevel, MeasReturnType
 from qiskit.validation.jsonschema import SchemaValidationError
+from qiskit.providers import BaseBackend
+from qiskit.pulse.channels import PulseChannel
 
 
 # TODO: parallelize over the experiments (serialize each separately, then add global header/config)
-def assemble(experiments,
-             backend=None,
-             qobj_id=None, qobj_header=None,
-             shots=None, memory=False, max_credits=None, seed_simulator=None,
-             qubit_lo_freq=None, meas_lo_freq=None,
-             qubit_lo_range=None, meas_lo_range=None,
-             schedule_los=None, meas_level=MeasLevel.CLASSIFIED,
-             meas_return=MeasReturnType.AVERAGE, meas_map=None,
-             memory_slot_size=100, rep_time=None, parameter_binds=None,
-             parametric_pulses=None,
-             **run_config):
+def assemble(experiments: Union[QuantumCircuit, List[QuantumCircuit]],
+             backend: Optional[BaseBackend] = None,
+             qobj_id: Optional[str] = None, qobj_header: Optional[Union[QobjHeader, Dict]] = None,
+             shots: Optional[int] = None, memory: Optional[bool] = False,
+             max_credits: Optional[int] = None,
+             seed_simulator: Optional[int] = None,
+             qubit_lo_freq: Optional[List[int]] = None, meas_lo_freq: Optional[List[int]] = None,
+             qubit_lo_range: Optional[List[int]] = None, meas_lo_range: Optional[List[int]] = None,
+             schedule_los: Optional[Union[List[Union[Dict[PulseChannel, float], LoConfig]],
+                                          Union[Dict[PulseChannel, float], LoConfig]]] = None,
+             meas_level: Union[int, MeasLevel] = MeasLevel.CLASSIFIED,
+             meas_return: Union[str, MeasReturnType] = MeasReturnType.AVERAGE,
+             meas_map: Optional[List[List[Qubit]]] = None,
+             memory_slot_size: int = 100, rep_time: Optional[float] = None,
+             parameter_binds: Optional[List[Dict[Parameter, float]]] = None,
+             parametric_pulses: Optional[List[str]] = None,
+             **run_config: Dict) -> Qobj:
     """Assemble a list of circuits or pulse schedules into a Qobj.
 
     This function serializes the payloads, which could be either circuits or schedules,
     to create Qobj "experiments". It further annotates the experiment payload with
     header and configurations.
 
     Args:
-        experiments (QuantumCircuit or list[QuantumCircuit] or Schedule or list[Schedule]):
+        experiments:
             Circuit(s) or pulse schedule(s) to execute
 
-        backend (BaseBackend):
+        backend:
             If set, some runtime options are automatically grabbed from
-            backend.configuration() and backend.defaults().
+            `backend.configuration()` and `backend.defaults()`.
```

Comment:
for rst, double backticks renders as code

Reply:
(I see below that you know that, maybe just a typo here?)

Reply:
Oops, yeah must have forgotten the double backticks in this file (I think I realized I forgot them in other parts of the file as well). Will fix it.